src/ContainerTerminal.jsx

```diff
@@ -0,0 +1,187 @@
+
+    componentWillUpdate(nextProps, nextState) {
+        if (nextState.cols !== this.state.cols) {
+            this.state.term.resize(nextState.cols, 24);
+            // TODO use something different than spawn (cockpit.file?) so also can read current size
+            cockpit.spawn(["sh", "-c", "echo '1 24 " + nextState.cols.toString() + "'>" + this.state.control_channel], { superuser: true });
+        }
+    }
+
+    connectChannel() {
+        let self = this;
+        if (self.state.channel)
+            return;
+
+        if (self.props.containerStatus !== "running") {
+            let message = _("Container is not running");
+            this.setState({ errorMessage: message });
+            return;
+        }
+
+        varlink.call(utils.PODMAN_ADDRESS, "io.podman.GetAttachSockets", { name: this.state.container })
+                .then(out => {
+                    let opts = {
+                        payload: "packet",
+                        unix: out.sockets.io_socket,
+                        superuser: "require",
+                        binary: false
+                    };
+
+                    let channel = cockpit.channel(opts);
+                    channel.wait()
+                            .then(() => {
+                                // Show the terminal. Once it was shown, do not show it again but reuse the previous one
+                                if (!this.state.opened) {
+                                    this.state.term.open(this.refs.terminal);
+                                    this.setState({ opened: true });
+                                }
+
+                                self.state.term.on('data', function(data) {
+                                    if (self.state.channel)
+                                        self.state.channel.send(data);
+                                });
+
+                                channel.addEventListener("message", function(event, data) {
+                                    self.state.term.write(data.substring(1)); // Drop first character which is marking stdin/stdout/stderr
+                                });
+                                channel.addEventListener('close', this.onChannelClose.bind(this));
+                                window.addEventListener('resize', this.onWindowResize);
+
+                                channel.send(String.fromCharCode(12)); // Send SIGWINCH to show prompt on attaching
+                                this.setState({ channel: channel, control_channel: out.sockets.control_socket, errorMessage: "" });
+                                this.onWindowResize();
+                            })
+                            .catch(e => {
+                                let message = cockpit.format(_("Could not open channel: $0"), e.problem);
+                                if (e.problem === "not-supported")
+                                    message = _("Newer version of cockpit is required in order to show terminal");
```

Comment:
Maybe it's better to say _web console_ here?

Reply:
Agreed. We never do (and never should) refer to "Cockpit" within Cockpit.

"Upgrade to newer version of the Web Console for an interactive terminal." is probably better text. It communicates the intent of the tab, states the problem, and includes a solution.

Backing up a bit: In what cases is this message necessary? Can we remove the tab completely for unsupported versions? Otherwise, it's basically UI leftovers that don't serve a purpose. None of the containers would have a console, yet they would all have a vestigial tab that teases one.

It's probably exceptionally rare that someone would run a version of Cockpit that's too old to have support for the terminal in the Podman page and also run the Podman page within Cockpit. (Of course, we should handle this exception so there's no outright _error_ if this happens, but I'm not sure we need a message here. Do we?)

Reply:
> Backing up a bit: In what cases is this message necessary? Can we remove the tab completely for unsupported versions? Otherwise, it's basically UI leftovers that don't serve a purpose. None of the containers would have a console, yet they would all have a vestigial tab that teases one.

This is necessary, because we need to be able to connect to a different type of socket - for which support has been implemented just recently. As you said, this is very corner and rare case and it is going to be just more rare as time goes by. However, we still need to be able to tell users why it is not working. 
There has been discussion if we should  not show the tab and two major reasons for showing the tab with the message were:
1) It is less confusing for users - when they expect to have console, they are going to ask why it is not there - such as they read release notes but they don't see it there. (for example see this https://github.com/cockpit-project/cockpit/issues/11086 )
2) It is very tricky to implement nicely - there is no nice way how to find out if cockpit supports this feature already - other than trying to connect to some socket. That would mean the only reasonable way how to determine if we can do it would be to try to connect to some socket and see if we were successful.